src/components/FullscreenModal/hooks/useStickyObserver.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useRef } from 'react';
+import debounce from 'lodash.debounce';
+
+export const useStickyObserver = (
+  modalRef: React.MutableRefObject<HTMLElement>,
+  elementRef: React.MutableRefObject<HTMLElement>,
+  isInView: () => void,
+): void => {
+  const observerRef = useRef(null);
+  const initialRender = useRef(true);
+  const scrollListener = debounce(isInView, 100);
+  const resizeListener = debounce(isInView, 200);
```

Comment:
This approach for listeners is valid in useStickyFooter, but not going work here, since useEffect below will be run multiple times, so second time it will launched, the references to the listeners will be lost. So I'd use ref here instead and create listener in `useEffect`. Example:
```
const listeners = useRef(null);
...
useEffect(()=>{
    if(listeners.current != null) {
        currentModalRef.removeEventListener('scroll', listeners.current.scrollListener);
        window.removeEventListener('resize', listeners.current.resizeListener);
    }
    listeners.current = {resizeListener: debounce(isInView, 200), scrollListener:debounce(isInView, 100)};
    currentModalRef.addEventListener('scroll', listeners.current.scrollListener);
    window.addEventListener('resize',  listeners.current.resizeListener);    
});
```